crawl-ref/source/mutation.cc

```diff
@@ -788,30 +786,55 @@ static bool _accept_mutation(mutation_type mutat, bool ignore_weight = false)
     return x_chance_in_y(weight, 10);
 }
 
-static mutation_type _get_mut_with_use(mutflag mt)
+static mutation_type _get_mut_with_use(mutflag mt, bool prefer_existing)
 {
     const int tweight = lookup(total_weight, mt, 0);
     ASSERT(tweight);
 
     int cweight = random2(tweight);
+
+    // Perform an initial loop to see if we can increase an existing mutation
+    // Since tweight includes weights of mutations we won't use, this is
+    // probably (definitely) bad, biased math. XXX please fix me.
+    if (prefer_existing)
```

Comment:
the logic of this block is totally off. I don't know how to make it smarter though...